testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -24,612 +23,419 @@ import (
 )
 
 var (
-	errSszCast             = errors.New("ssz response is not a byte array")
-	errJsonCast            = errors.New("json response has wrong structure")
-	errEmptyPrysmData      = errors.New("prysm data is empty")
-	errEmptyLighthouseData = errors.New("lighthouse data is empty")
+	errSszCast   = errors.New("SSZ response is not a byte array")
+	errEmptyData = errors.New("data is empty")
 )
 
-type metadata struct {
-	start            primitives.Epoch
-	basepath         string
-	params           func(encoding string, currentEpoch primitives.Epoch) []string
-	requestObject    interface{}
-	prysmResps       map[string]interface{}
-	lighthouseResps  map[string]interface{}
-	customEvaluation func(interface{}, interface{}) error
+const (
+	msgWrongJson = "JSON response has wrong structure, expected %T, got %T"
+)
+
+type meta interface {
+	getBasePath() string
+	sszEnabled() bool
+	enableSsz()
+	getSszResp() []byte
+	setSszResp(resp []byte)
+	getStart() primitives.Epoch
+	setStart(start primitives.Epoch)
+	getReq() interface{}
+	setReq(req interface{})
+	getPResp() interface{}
+	getLResp() interface{}
+	getParams(epoch primitives.Epoch) []string
+	setParams(f func(primitives.Epoch) []string)
+	getCustomEval() func(interface{}, interface{}) error
+	setCustomEval(f func(interface{}, interface{}) error)
+}
+
+type metadata[Resp any] struct {
+	basePath   string
+	ssz        bool
+	start      primitives.Epoch
+	req        interface{}
+	pResp      *Resp
+	lResp      *Resp
+	sszResp    []byte
+	params     func(currentEpoch primitives.Epoch) []string
+	customEval func(interface{}, interface{}) error
```

Comment:
I don't think this can be made generic because it would require the interface to be generic as well. And if I make the interface generic, then `var requests = map[string]meta{}` won't work because a generic interface will be required. But since each item has a different response type, I won't be able to create a map of `meta` items.